gomock/argument_captor.go

```diff
@@ -0,0 +1,40 @@
+package gomock
+
+// ArgumentCaptor is a struct that composes a Matcher, and extends it by storing given arguments in the values slice.
+type ArgumentCaptor struct {
+	Matcher
```

Comment:
I think `m Matcher` would suit us better here. If you ever need to do anything fancy /w reflection or anything working with embedded fields can be hard.